### PR TITLE
[FLINK-7810] Switch from custom Flakka to Akka 2.4.x 

### PR DESCRIPTION
**Pull request by aljoscha**
## What is the purpose of the change

Drop support for Scala 2.10 and then update to a newer Akka version. Before, we were forced to stay on our custom Flakka 2.3.x version with back ports because newer Akka does not support Scala 2.10.

### CODE REVIEW

**Comment by bowenli86** [comment]
wonderful! +1

File: docs/start/building.md [hunk 1]
```diff
@@ -115,19 +115,7 @@ The `-Pvendor-repos` activates a Maven [build profile](http://maven.apache.org/g
...
-mvn clean install -DskipTests -Pscala-2.10
-~~~
-
-To build against custom Scala versions, you need to define new custom build profile that will override *scala.version* and *scala.binary.version* values.
-
-Flink is developed against Scala *2.11* and tested additionally against Scala *2.10*. These two versions are known to be compatible. Earlier versions (like Scala *2.9*) are *not* compatible.
-
-Newer versions may be compatible, depending on breaking changes in the language features used by Flink, and the availability of Flink's dependencies in those Scala versions. The dependencies written in Scala include for example *Kafka*, *Akka*, *Scalatest*, and *scopt*.
-
+**By default, Flink is built with the Scala 2.11**.
```
Comment (GJL) [inline docs/start/building.md]: I would drop *the*, i.e., 
> **By default, Flink is built with Scala 2.11**.

File: docs/start/building.md [hunk 2]
```diff
@@ -115,19 +115,7 @@ The `-Pvendor-repos` activates a Maven [build profile](http://maven.apache.org/g
...
-**By default, Flink is built with the Scala 2.11**. To build Flink with Scala *2.10*, you can change the default Scala *binary version* by using *scala-2.10* build profile:
-
-~~~bash
-# Build with Scala version 2.10
-mvn clean install -DskipTests -Pscala-2.10
-~~~
-
-To build against custom Scala versions, you need to define new custom build profile that will override *scala.version* and *scala.binary.version* values.
-
-Flink is developed against Scala *2.11* and tested additionally against Scala *2.10*. These two versions are known to be compatible. Earlier versions (like Scala *2.9*) are *not* compatible.
```
Comment (EronWright) [inline docs/start/building.md]: I read this paragraph to imply that Flink is still compatible with 2.10, but is that accurate given the use of Akka 2.4?

**Comment by EronWright** [comment]
Thanks for doing this, we need Akka 2.4 badly for improved SSL support.

File: docs/start/building.md (hunk 1 above)
Comment (aljoscha) [reply docs/start/building.md]: fixing

File: docs/start/building.md (hunk 2 above)
Comment (aljoscha) [reply docs/start/building.md]: Yes, I don't explicitly say what we're compatible with now. In fact we only support 2.11 now, supporting 2.12 is a bit harder than I though.

File: .travis.yml [hunk 3]
```diff
@@ -61,31 +61,6 @@ matrix:
         - TEST="misc"
         - PROFILE="-Dhadoop.version=2.8.0"
         - CACHE_NAME=JDK8_H280_M
-    - jdk: "openjdk8"
```
Comment (StephanEwen) [inline .travis.yml]: How about retaining these Travis profiles for now for the Hadoop and JDK versions?
Just bump the Scala dependency.

File: flink-queryable-state/flink-queryable-state-java/pom.xml [hunk 4]
```diff
@@ -110,8 +110,8 @@ under the License.
 		</dependency>
 
 		<dependency>
-			<groupId>com.data-artisans</groupId>
-			<artifactId>flakka-testkit_${scala.binary.version}</artifactId>
+			<groupId>com.typesafe.akka</groupId>
```
Comment (StephanEwen) [inline flink-queryable-state/flink-queryable-state-java/pom.xml]: I could not find any use of akka testkit in the queryable state client (and there should not be, design wise).

Was this just copy/paste forwarded and can be dropped?

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskManagerComponentsStartupShutdownTest.java [hunk 5]
```diff
@@ -202,8 +202,6 @@ protected void run() {
 			assertTrue(ioManager.isProperlyShutDown());
 			assertTrue(memManager.isShutdown());
 		} finally {
-			TestingUtils.stopActorsGracefully(Arrays.asList(jobManager, taskManager));
```
Comment (StephanEwen) [inline flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskManagerComponentsStartupShutdownTest.java]: Is this now happening automatically?

File: flink-queryable-state/flink-queryable-state-java/pom.xml (hunk 4 above)
Comment (aljoscha) [reply flink-queryable-state/flink-queryable-state-java/pom.xml]: There was a test in there that was using Akka but I think that's not needed anymore. Will remove.

On a side node, currently, the queryable-state-client module contains both the client and server implementations (that's why there was the akka test) while flunk-runtime only has interfaces. Fully tweezing the client and server apart is still in progress.

File: .travis.yml (hunk 3 above)
Comment (aljoscha) [reply .travis.yml]: You're right, of course. Undropping ...

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskManagerComponentsStartupShutdownTest.java (hunk 5 above)
Comment (aljoscha) [reply flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskManagerComponentsStartupShutdownTest.java]: If the test succeeds, they will have been shutdown (the previous asserts also assert (hehe) this.

It seems the previous Akka version didn't mind another shutdown attempt but 2.4 fails with an exception when you try to shutdown after already being shut down.

**Comment by StephanEwen** [comment]
I think this is good now, +1

**Pull request closed without merging**